service/history/timerBuilder.go

```diff
@@ -303,7 +303,7 @@ func (tb *timerBuilder) loadActivityTimers(msBuilder *mutableStateBuilder) {
 					td := &timerDetails{
 						TimerSequenceID: TimerSequenceID{VisibilityTimestamp: heartBeatExpiry},
 						ActivityID:      v.ScheduleID,
-						EventID:         v.StartedID,
+						EventID:         v.ScheduleID,
 						TimeoutType:     w.TimeoutTypeHeartbeat,
 						TimeoutSec:      v.HeartbeatTimeout,
 						TaskCreated:     (v.TimerTaskStatus & TimerTaskStatusCreatedHeartbeat) != 0}
```

Comment:
The problem is after first heartbeat timer is created, the bit for heartbeat timer will be set, and after that this TaskCreated will always be true so firstActivityTimerTask() will not return the timer task and it won't be created.

Reply:
@yiminc 
if my understanding is correct, https://github.com/uber/cadence/blob/master/service/history/timerQueueActiveProcessor.go#L307 will take care of that?

Reply:
i actually have some doubt:
here: https://github.com/uber/cadence/pull/658/files#diff-b3d25d1c01ad6e7393fd3958caea1333R287
the start event ID is actually checked
so when the timer detail is created (by loading activity into memory), we are sure that the start event ID is a valid event ID

Reply:
Yes, I think this fix works.

Reply:
The problem is the startedID is buffered event id which changed to real start event after it is flushed.